# Import

In [1]:
import numpy as np
import pandas as pd

# load data
df = pd.read_csv('train.csv')

df

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S
...,...,...,...,...,...,...,...,...,...,...,...,...
886,887,0,2,"Montvila, Rev. Juozas",male,27.0,0,0,211536,13.0000,NaN,S
887,888,1,1,"Graham, Miss. Margaret Edith",female,19.0,0,0,112053,30.0000,B42,S
888,889,0,3,"Johnston, Miss. Catherine Helen ""Carrie""",female,NaN,1,2,W./C. 6607,23.4500,NaN,S
889,890,1,1,"Behr, Mr. Karl Howell",male,26.0,0,0,111369,30.0000,C148,C


# Basic Information

In [9]:
print(df.info())

print(df.head())

print(df.tail(3))

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 118.9 KB
None
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                    

# Basic Statistics

In [15]:
print(df.describe())

print(f"Age mean: {df['Age'].mean()}")
print(f"Age medium: {df['Age'].median()}")
print(f"Age mode: {df['Age'].mode()[0]}")

print(f"Fare mean: {df['Fare'].mean()}")
print(f"Fare median: {df['Fare'].median()}")
print(f"Fare mode: {df['Fare'].mode()[0]}")

       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  
Age mean: 29.69911764705882
Age medium: 28.0
Age mode: 24.0
Fare mean: 32.2042079

# Check Invalid Data

In [16]:
df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

#### Analysis: According to the NaN check, the information of 'Age' and 'Cabin' are lost, which is possibly because they are complicated to survey.

# Single Variable Analysis

### Age

In [39]:
age_ascending = pd.DataFrame(df.sort_values(by = 'Age')['Age'].dropna())
print(age_ascending)

       Age
803   0.42
755   0.67
644   0.75
469   0.75
78    0.83
..     ...
116  70.50
493  71.00
96   71.00
851  74.00
630  80.00

[714 rows x 1 columns]


In [46]:
adult = age_ascending[age_ascending['Age'] > 18]
print(adult)
print(f"adult: {len(adult)/len(age_ascending) * 100}%")

print(f"non-adult: {(1-len(adult)/len(age_ascending)) * 100}%")

      Age
226  19.0
136  19.0
27   19.0
715  19.0
291  19.0
..    ...
116  70.5
493  71.0
96   71.0
851  74.0
630  80.0

[575 rows x 1 columns]
adult: 80.53221288515407%
non-adult: 19.467787114845937%


In [43]:
infants = age_ascending[age_ascending['Age'] <= 3]
print(infants)
print(f"{len(infants)/len(age_ascending) * 100}%")

      Age
803  0.42
755  0.67
644  0.75
469  0.75
78   0.83
831  0.83
305  0.92
827  1.00
381  1.00
164  1.00
183  1.00
386  1.00
172  1.00
788  1.00
642  2.00
7    2.00
530  2.00
297  2.00
824  2.00
205  2.00
119  2.00
340  2.00
479  2.00
16   2.00
348  3.00
407  3.00
374  3.00
261  3.00
43   3.00
193  3.00
4.201680672268908%


In [44]:
elderly = age_ascending[age_ascending['Age'] >= 60]
print(elderly)
print(f"{len(elderly)/len(age_ascending) * 100}%")

      Age
694  60.0
684  60.0
587  60.0
366  60.0
170  61.0
326  61.0
625  61.0
252  62.0
829  62.0
570  62.0
555  62.0
275  63.0
483  63.0
438  64.0
545  64.0
54   65.0
280  65.0
456  65.0
33   66.0
672  70.0
745  70.0
116  70.5
493  71.0
96   71.0
851  74.0
630  80.0
3.6414565826330536%


#### Analysis: 'pre-adult' accounts for close to 20%. 'Infants' accounts for 4.20%. 'elderly' accounts for 3.64%. The youngest is only 0.42 years old and the oldest is 80 years old.

### Sex

In [61]:
male = df[df['Sex'] == 'male']
female = df[df['Sex'] == 'female']

print(len(male))
print(len(female))
print(f"{len(male)/len(df) * 100}%")

577
314
64.75869809203144%


#### Analysis: 64.76% passengers are male

### Cabin

In [64]:
Cabin = pd.DataFrame(df['Cabin'].dropna())

print(Cabin)

           Cabin
1            C85
3           C123
6            E46
10            G6
11          C103
..           ...
871          D35
872  B51 B53 B55
879          C50
887          B42
889         C148

[204 rows x 1 columns]


### Survived

In [66]:
Survived = pd.DataFrame(df['Survived'])
print(Survived)

     Survived
0           0
1           1
2           1
3           1
4           0
..        ...
886         0
887         1
888         0
889         1
890         0

[891 rows x 1 columns]


In [68]:
Survived.value_counts()

Survived
0           549
1           342
Name: count, dtype: int64

#### Analysis: Most passengers are not survived.

# Inference

### Sex & Survival

In [77]:
male_survived = pd.DataFrame(male['Survived'])
print(male_survived)

male_survived_count = male_survived.value_counts()
print(male_survived_count)

male_survived_rate = str(109 / len(male_survived) * 100) + '%'
print(male_survived_rate)

     Survived
0           0
4           0
5           0
6           0
7           0
..        ...
883         0
884         0
886         0
889         1
890         0

[577 rows x 1 columns]
Survived
0           468
1           109
Name: count, dtype: int64
18.890814558058924%


In [80]:
female_survived = pd.DataFrame(female['Survived'])
print(female_survived)

female_survived_count = female_survived.value_counts()
print(female_survived_count)

female_survived_rate = str(233 / len(female_survived) * 100) + '%'
print(female_survived_rate)

     Survived
1           1
2           1
3           1
8           1
9           1
..        ...
880         1
882         0
885         0
887         1
888         0

[314 rows x 1 columns]
Survived
1           233
0            81
Name: count, dtype: int64
74.20382165605095%


#### Analysis: the survival rate of females is much higher than that of males.

### Age & Survival

In [85]:
non_adult = pd.DataFrame(df[df['Age'] < 18])

adult = pd.DataFrame(df[df['Age'] >= 18])

non_adult_survived = non_adult['Survived'].value_counts()

adult_survived = adult['Survived'].value_counts()

print(non_adult_survived)
print(adult_survived)

non_adult_survived_rate = 61/len(non_adult) * 100
adult_survived_rate = 229/len(adult) * 100

print(f'{non_adult_survived_rate}%')
print(f'{adult_survived_rate}%')

Survived
1    61
0    52
Name: count, dtype: int64
Survived
0    372
1    229
Name: count, dtype: int64
53.98230088495575%
38.10316139767055%


#### Analysis: The rate of survival of minors is significantly higher than that of adults.

# Conclusion: Female underage is the easiest to survive.